src/context.ts

```diff
@@ -185,39 +185,36 @@ const checker: StaticHas = {
         };
     },
     command(command) {
-        const hasEntities = checker.filterQuery(":entities:bot_command");
-        const atCommands = new Set<string>();
-        const noAtCommands = new Set<string>();
-        toArray(command).forEach((cmd) => {
+        const hasEntities = checker.filterQuery("::bot_command");
+        const noAtCommands = new Set<string>(toArray(command ?? []));
+        for (const cmd of noAtCommands) {
             if (cmd.startsWith("/")) {
                 throw new Error(
                     `Do not include '/' when registering command handlers (use '${
                         cmd.substring(1)
                     }' not '${cmd}')`,
                 );
             }
-            const set = cmd.includes("@") ? atCommands : noAtCommands;
-            set.add(cmd);
-        });
+        }
         return <C extends Context>(ctx: C): ctx is CommandContext<C> => {
             if (!hasEntities(ctx)) return false;
             const msg = ctx.message ?? ctx.channelPost;
-            const txt = msg.text ?? msg.caption;
-            return msg.entities.some((e) => {
+            const txt = msg.text ?? msg.caption ?? "";
+            const entities = msg.entities ?? msg.caption_entities;
+            return entities.some((e) => {
                 if (e.type !== "bot_command") return false;
                 if (e.offset !== 0) return false;
                 const cmd = txt.substring(1, e.length);
-                if (noAtCommands.has(cmd) || atCommands.has(cmd)) {
-                    ctx.match = txt.substring(cmd.length + 1).trimStart();
-                    return true;
+                let index = cmd.indexOf("@");
+                if (index === -1) {
+                    index = Infinity;
```

Comment:
This is a big of a hack, isn't it?

Reply:
Not if `.substring()` was replaced with `.slice()`, or
```suggestion
                    index = cmd.length;
```
`undefined` would require adding a type annotation.